app/javascript/angular/code/services/_fixed_sessions.js

```diff
@@ -82,31 +78,39 @@ export const fixedSessions = (
           };
           params.update({ prevMapPosition: prevMapPosition });
           map.fitBoundsWithBottomPadding(calculateBounds(sensors, sessionData));
-          this.drawMarkersWithLabel(sessionData, sensors.selectedSensorName());
+          this.drawSelectedSession(sessionData);
         }
       };
       params.update({ selectedSessionIds: [id] });
       this._selectSession(id, fitBounds);
     },
 
     reSelectSession: function(id) {
-      const noop = sessionData => {
+      const callback = sessionData => {
         if (!sessionData.is_indoor) {
-          this.drawMarkersWithLabel(sessionData, sensors.selectedSensorName());
+          map.fitBoundsWithBottomPadding(calculateBounds(sensors, sessionData));
+          this.drawSelectedSession(sessionData);
         }
       };
-      this._selectSession(id, noop);
+      this._selectSession(id, callback);
     },
 
+    drawSelectedSession: function(sessionData) {
+      if (params.isStreaming()) {
+        this.drawMarkersWithLabel(sessionData, sensors.selectedSensorName());
+      } else {
+        this.drawMarkersWithoutLabel(sessionData);
+      }
+    },
     _selectSession: function(id, callback) {
       var sensorId = sensors.selectedId();
       var sensor = sensors.sensors[sensorId] || {};
       var sensorName = sensor.sensor_name;
       if (!sensorName) return;
       $http
-        .get("/api/realtime/sessions/" + id, {
```

Comment:
Can we remove the old endpoint?

Reply:
Probably, I was planning to look into that in a separate PR.